scalyr_agent/tests/copying_manager_test.py

```diff
@@ -1022,6 +1046,66 @@ def test_start_without_active_checkpoint(self):
         self.assertEquals("Third line", lines[0])
         self.assertEquals("Fourth line", lines[1])
 
+    def test_start_with_bad_checkpoint(self):
+        controller = self.__create_test_instance()
+        previous_root_dir = os.path.dirname(self.__test_log_file)
+
+        self.__append_log_lines("First line", "Second line")
+        (request, responder_callback) = controller.wait_for_rpc()
+        lines = self.__extract_lines(request)
+
+        self.assertEquals(2, len(lines))
+        self.assertEquals("First line", lines[0])
+        self.assertEquals("Second line", lines[1])
+        controller.stop()
+
+        self.__append_log_lines("Third line", "Fourth line")
```

Comment:
Add some comments explaining what you are looking for here.  For example, "In the case of a bad checkpoint file, the agent should just pretend the checkpoint file does not exist and start reading the logfiles from the end.  In this case, that means lines three and four will be skipped."